samcli/lib/build/app_builder.py

```diff
@@ -521,11 +526,10 @@ def _build_function(  # pylint: disable=R1710
                 # By default prefer to build in-process for speed
                 if self._container_manager:
                     image = None
-                    if self._build_images is not None:
-                        if function_name in self._build_images:
-                            image = self._build_images[function_name]
-                        elif None in self._build_images:
-                            image = self._build_images[None]
+                    if function_name in self._build_images:
+                        image = self._build_images.get(function_name)
+                    elif None in self._build_images:
+                        image = self._build_images.get(None)
```

Comment:
the entire code block (including the `image = None`) can be simplified as

```py
image = self._build_images.get(function_name, self._build_images.get(None))
```

Reply:
I feel that this logic here is simple and I don't think the code is over-complicated, I would argue for keeping it this way? For better readability

Reply:
duplicate from the one above, resolving